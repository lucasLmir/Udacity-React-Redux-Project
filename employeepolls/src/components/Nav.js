import { Link } from "react-router-dom";
import { connect } from "react-redux";

const Nav = (props) => (
  <nav className="nav">
    <ul>
      <li>
        <Link to="/">Home</Link>
      </li>
      <li>
        <Link to="/new">Ask a Question</Link>
      </li>
      <li>
        <Link to="/leaderboard">Leader Board</Link>
      </li>
      <li className="right">
        <Link to="/login">
          <span data-testid="userName">
            {props.name === undefined ? "Login" : props.name}
          </span>
        </Link>
      </li>
      <li>
        <Link to="/login">
          <img
            data-testid="avatarImg"
            hidden={props.name === undefined}
            alt={`avatar of ${props.name}`}
            src={props.avatar}
            className="small-avatar"
          />
        </Link>
      </li>
    </ul>
  </nav>
);

const mapStateToProps = ({ authedUser, users }) => {
  const avatar = users[authedUser]?.avatarURL;
  const name = users[authedUser]?.name;

  return {
    name,
    avatar,
  };
};

export default connect(mapStateToProps)(Nav);
