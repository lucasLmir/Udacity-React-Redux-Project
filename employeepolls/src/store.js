import reducer from "./reducers";
import { createStore } from "redux";
import middleware from "./middleware";

export const store = createStore(reducer, middleware);