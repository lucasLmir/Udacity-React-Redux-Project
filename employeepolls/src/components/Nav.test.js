import { render } from "@testing-library/react";
import Nav from "./Nav";
import { store } from "../store";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import { setAuthedUser } from "../actions/authedUser";
import { receiveUsers } from "../actions/users";

describe("Nav", () => {
  it("is expected to render the Nav component", () => {
    const component = render(
      <Provider store={store}>
        <Router>
          <Nav />
        </Router>
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
  it("The user name and avatar image in the Navbar, it is not expected when no user is logged in.", () => {
    const component = render(
      <Provider store={store}>
        <Router>
          <Nav />
        </Router>
      </Provider>
    );

    expect(component.getByTestId("userName").innerHTML).toBe("Login");
    expect(component.queryByTestId("avatarImg")).not.toBeVisible();
  });
  it("The name of the user and avatar image should appear in the Navbar when the user is authorized.", async () => {
    const users = {
      mtsamis: {
        id: 'mtsamis',
        name: 'Mike Tsamis',
        avatarURL: "https://www.shutterstock.com/image-vector/young-smiling-man-adam-avatar-600w-2107967969.jpg",
      },
    }
    await store.dispatch(receiveUsers(users));
    await store.dispatch(setAuthedUser("mtsamis"));
    const component = render(
      <Provider store={store}>
        <Router>
          <Nav />
        </Router>
      </Provider>
    );

    expect(component.getByTestId("userName").innerHTML).toBe("Mike Tsamis");
    expect(component.queryByTestId("avatarImg")).toBeVisible();
  });
});
