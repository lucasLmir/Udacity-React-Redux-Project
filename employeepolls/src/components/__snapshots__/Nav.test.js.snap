// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav is expected to render the Nav component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="nav"
      >
        <ul>
          <li>
            <a
              href="/"
            >
              Home
            </a>
          </li>
          <li>
            <a
              href="/add"
            >
              Ask a Question
            </a>
          </li>
          <li>
            <a
              href="/leaderboard"
            >
              Leader Board
            </a>
          </li>
          <li
            class="mid-li"
          />
          <li>
            <a
              hidden=""
              href="/"
            >
              LogOut
            </a>
          </li>
          <li>
            <a
              href="/login"
            >
              <span
                data-testid="userName"
              >
                Login
              </span>
            </a>
          </li>
          <li>
            <a
              href="/login"
            >
              <img
                alt="avatar of undefined"
                class="small-avatar"
                data-testid="avatarImg"
                hidden=""
              />
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="nav"
    >
      <ul>
        <li>
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="/add"
          >
            Ask a Question
          </a>
        </li>
        <li>
          <a
            href="/leaderboard"
          >
            Leader Board
          </a>
        </li>
        <li
          class="mid-li"
        />
        <li>
          <a
            hidden=""
            href="/"
          >
            LogOut
          </a>
        </li>
        <li>
          <a
            href="/login"
          >
            <span
              data-testid="userName"
            >
              Login
            </span>
          </a>
        </li>
        <li>
          <a
            href="/login"
          >
            <img
              alt="avatar of undefined"
              class="small-avatar"
              data-testid="avatarImg"
              hidden=""
            />
          </a>
        </li>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
